LoRaEngine/test/LoRaWan.Test.Shared/IntegrationTestFixtureBase.Asserts.cs

```diff
@@ -135,6 +135,52 @@ public async Task AssertNetworkServerModuleLogExistsAsync(Func<string, bool> pre
             }
         }
 
+        /// <summary>
+        /// Check for The timing between an upstream message and a downstream one.
+        /// The method stop at first occurence found and is not expected to perform against multiple DS/US messages.
+        /// </summary>
+        public async Task CheckAnswerTimingAsync(uint rxDelay, bool isSecondWindows)
+        {
+            var upstreamMessageTimingResult = await this.TryFindMessageTimeAsync("rxpk");
+            var downstreamMessageTimingResult = await this.TryFindMessageTimeAsync("txpk");
+
+            if (upstreamMessageTimingResult.Success && upstreamMessageTimingResult.Success)
+            {
+                Assert.Equal(rxDelay + (isSecondWindows ? 1000000 : 0), downstreamMessageTimingResult.Result - upstreamMessageTimingResult.Result);
+            }
+            else
+            {
+                Assert.True(false, "Could not procees with timestamp message search.");
+            }
+        }
+
+        /// <summary>
+        /// Helper method to find the time of the message that contain the message argument.
+        /// </summary>
+        async Task<FindTimeResult> TryFindMessageTimeAsync(string message)
+        {
+            var log = await this.SearchUdpLogs(x => x.Contains(message));
+            int timeIndexStart = log.FoundLogResult.IndexOf(@"""tmst"":") + 7;
+            int timeIndexStop = log.FoundLogResult.IndexOf(",", timeIndexStart);
+            uint parsedValue = 0;
+            bool success = false;
+            if (timeIndexStart > 0 && timeIndexStop > 0)
+            {
+                if (uint.TryParse(log.FoundLogResult.Substring(timeIndexStart, timeIndexStop - timeIndexStart), out parsedValue))
+                {
+                    success = true;
+                }
+            }
+
+            var result = new FindTimeResult()
```

Comment:
directly return the result instead of assigning first? return new FindTimeResult ...